packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,264 @@
+  children: React.ReactNode;
+
+  /**
+   * Determines the open state of the modal
+   *
+   * default: `false`
+   */
+  open: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the open state of the Modal.
+   *
+   */
+  setOpen?: (open: boolean) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  open
+  size="large"
+  setOpen={setOpen}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >
+  Modal content!
+</Modal>
+```
+ * @param props.open Boolean to describe whether or not Modal is open.
+ * @param props.size String to determine size of Modal. ['small', 'default', 'large']
+ * @param props.setOpen Callback to change the open state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  open = false,
+  size = ModalSize.Default,
+  setOpen = () => {},
+  shouldClose = () => true,
+  children,
+  className,
+  ...rest
+}: ModalProps) {
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    // Don't close modal if shouldClose returns false or no setOpen callback is passed
+    if (!setOpen || !shouldClose()) {
+      return;
+    }
+
+    setOpen(false);
+  }, [setOpen, shouldClose]);
+
+  const handleBackdropClick = (e: React.SyntheticEvent) => {
+    if (!scrollContainerRef.current) {
+      return;
+    }
+
+    if (e.target === scrollContainerRef.current) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.keyCode === escapeKey) {
+        handleClose();
+      }
+    },
+    [handleClose],
+  );
+
+  useEventListener('keydown', handleEscape);
+
+  return (
+    <Transition in={open} timeout={500} mountOnEnter unmountOnExit>
+      {(state: string) => (
+        <Portal>
+          <div
+            {...rest}
+            onClick={handleBackdropClick}
+            className={cx(backdrop, {
+              [visibleBackdrop]: state === 'entered',
+            })}
+          >
+            <div className={scrollContainer} ref={scrollContainerRef}>
+              <div
+                className={cx(
+                  modalContentStyle,
+                  modalSizes[size],
+                  {
+                    [visibleModalContentStyle]: state === 'entered',
+                  },
+                  className,
+                )}
+                tabIndex={-1}
```

Comment:
Let's get rid of this. It's actually making the modal highlight as if it were focused when it's clicked.

Reply:
👍